include/wil/registry_basic.h

```diff
@@ -518,19 +222,31 @@ namespace wil
             }
 
 #if defined(__WIL_OLEAUTO_H_)
+            inline void* get_buffer(const BSTR& value) WI_NOEXCEPT
+            {
+                return value;
+            }
             inline void* get_buffer(const ::wil::unique_bstr& value) WI_NOEXCEPT
             {
                 return value.get();
             }
 #endif // #if defined(__WIL_OLEAUTO_H_)
 
-#if defined(__WIL_OLEAUTO_H_STL)
-            inline void* get_buffer(const ::wil::shared_bstr& value) WI_NOEXCEPT
+#if defined(__WIL_OBJBASE_H_)
+            inline void* get_buffer(const ::wil::unique_cotaskmem_string& value) WI_NOEXCEPT

```

Comment:
Did we explicitly remove shared_bstr?

Reply:
please see the current interfaces in how shared_bstr is supported. I spent a fair amount of time working through options with this.